src/components/AddPeopleAndRoles/OrgPerson.vue

```diff
@@ -0,0 +1,533 @@
+import { ConfirmDialog } from '@/components/dialogs'
+
+// Mixins
+import { EntityFilterMixin, CommonMixin } from '@/mixins'
+
+// Enums
+import { EntityTypes, Roles, IncorporatorTypes } from '@/enums'
+
+// Schemas
+import { addressSchema } from '@/schemas'
+
+@Component({
+  components: {
+    BaseAddress,
+    ConfirmDialog
+  }
+})
+export default class OrgPerson extends Mixins(EntityFilterMixin, CommonMixin) {
+   // Refs
+   $refs!: {
+    addPersonOrgForm: FormType,
+    mailingAddressNew: BaseAddressType,
+    deliveryAddressNew: BaseAddressType,
+    reassignCPDialog: DialogType
+  }
+
+  // Props
+  @Prop()
+  private initialValue!: OrgPersonIF
+
+  @Prop()
+  private activeIndex: number
+
+  @Prop()
+  private nextId: number
+
+  @Prop()
+  private existingCompletingParty: OrgPersonIF
+
+  // Data Properties
+  private orgPerson: OrgPersonIF
+  private addPersonOrgFormValid: boolean = true
+
+  // Address related properties
+  private inProgressMailingAddress:AddressIF
+  private inProgressDeliveryAddress:AddressIF
+  private inheritMailingAddress: boolean = true
+  private personAddressSchema: {} = addressSchema
+  private mailingAddressValid: boolean = false
+  private deliveryAddressValid: boolean = false
+  private reassignCompletingParty : boolean = false
+
+  // Roles
+  private isCompletingParty: boolean = false
+  private isIncorporator: boolean = false
+  private isDirector: boolean = false
+
+  readonly EntityTypes: {} = EntityTypes
+  readonly Roles: {} = Roles
+  readonly IncorporatorTypes: {} = IncorporatorTypes
+
+  // Rules
+  private readonly firstNameRules: Array<Function> = [
+    v => !!v || 'A first name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly middleNameRules: Array<Function> = [
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly lastNameRules: Array<Function> = [
+    v => !!v || 'A last name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly orgNameRules: Array<Function> = [
+    v => !!v || 'A firm name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  // Life cycle methods
+  private created (): void {
+    if (this.initialValue) {
+      this.orgPerson = { ...this.initialValue }
+      this.isDirector = this.orgPerson.roles.includes(Roles.DIRECTOR)
+      this.isIncorporator = this.orgPerson.roles.includes(Roles.INCORPORATOR)
+      this.isCompletingParty = this.orgPerson.roles.includes(Roles.COMPLETING_PARTY)
+      this.inProgressMailingAddress = { ...this.orgPerson.address.mailingAddress }
```

Comment:
Should any of these flags be watchers/computed properties so that they are automatically updated?

Reply:
There is no further update to the properties in this component other than the address which is handled in a different way altogether. So I dont think we need watchers here